File: clientbuilding/src/main/resources/static/js/service/typeService.js


/**
 * Service for Type
 **/

define(['require', 'angular'], function (require, angular) {
app.aService(clientbuilding.prefix + 'typeService', function($http, $rootScope) {
	
	// Create.
	this.create = function(type) {
		var serverUrl = clientbuilding.serverUrl + '/type/create';
		var request = {
				 method: 'POST',
				 url: serverUrl,
				 data: type
				}
		return $http(request);
	}
	
	// Update Lock.
	this.updateLock = function(id) {
		var serverUrl = clientbuilding.serverUrl + '/type/updateLock/' + id;
		var request = {
				 method: 'PUT',
				 url: serverUrl
				}
		return $http(request);
	}
	
	// Update Unlock.
	this.updateUnlock = function(id) {
		var serverUrl = clientbuilding.serverUrl + '/type/updateUnlock/' + id;
		var request = {
				 method: 'PUT',
				 url: serverUrl
				}
		return $http(request);
	}
	
	// Update.
	this.update = function(id, type) {
		var serverUrl = clientbuilding.serverUrl + '/type/update/' + id;
		var request = {
				 method: 'PUT',
				 url: serverUrl,
				 data: type
				}
		return $http(request);
	}
	
	// Update With Lock.
	this.updateWithLock = function(id, type) {
		var serverUrl = clientbuilding.serverUrl + '/type/updateWithLock/' + id;
		var request = {
				 method: 'PUT',
				 url: serverUrl,
				 data: type
				}
		return $http(request);
	}
	
	// Update For Delete.
	this.updateForDelete = function(id, version) {
		var serverUrl = clientbuilding.serverUrl + '/type/updateForDelete/' + id + '/' + version;
		var request = {
				 method: 'PUT',
				 url: serverUrl
				}
		return $http(request);
	}
	
	// Update For Delete With Lock.
	this.updateForDeleteWithLock = function(id, version) {
		var serverUrl = clientbuilding.serverUrl + '/type/updateForDeleteWithLock/' + id + '/' + version;
		var request = {
				 method: 'PUT',
				 url: serverUrl
				}
		return $http(request);
	}
	
	// Delete.
	this.delete = function(id) {
		var serverUrl = clientbuilding.serverUrl + '/type/delete/' + id;
		var request = {
				 method: 'DELETE',
				 url: serverUrl
				}
		return $http(request);
	}
	
	// Get by Id.
	this.getById = function(typeId) {
		var serverUrl = clientbuilding.serverUrl + '/type/getById/' + typeId;
		var request = {
				 method: 'GET',
				 url: serverUrl
				}
		return $http(request);
	}
	
	// List all user for select.
	this.listForSelectByScope = function(scope){
		var serverUrl = clientbuilding.serverUrl + '/type/listForSelectByScope/' + scope;
		var request = {
				method: 'GET',
				url: serverUrl
		}
		return $http(request);
	}
	
	// List by scope and page and filter.
	this.listWithCriteriasByScopeAndPage = function(scope, criterias, pageNo, pageSize, sorts) {
		var serverUrl = clientbuilding.serverUrl + '/type/listWithCriteriasByScopeAndPage/' + scope + '?page=' + pageNo + '&size=' + pageSize;
		if(typeof(sorts) !== 'undefined' && sorts.length > 0) {
			angular.forEach(sorts, function(sort) {
				serverUrl += '&' + sort;
			});
		}
		var request = {
				 method: 'POST',
				 url: serverUrl,
				 data: criterias
				}
		return $http(request);
	}

});

});

File: serverbuilding/src/main/java/com/redsun/server/building/service/TypeService.java

package com.redsun.server.building.service;

import java.io.IOException;
import java.util.List;
import java.util.Map;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import com.fasterxml.jackson.core.JsonParseException;
import com.fasterxml.jackson.databind.JsonMappingException;
import com.redsun.server.building.model.Type;
import com.redsun.server.building.model.common.SearchCriteria;

public interface TypeService {

	/**
	 * Saves (create or update) the given entity <br>
	 * Transactional operation ( begin transaction and commit )
	 * @param type
	 * @return
	 */
	Type save(Type type);

	/**
	 * Create the given entity <br>
	 * Transactional operation ( begin transaction and commit )
	 * @param type
	 * @return
	 */
	Type create(Type type) throws JsonParseException, JsonMappingException, IOException;

	/**
	 * Update the given entity <br>
	 * Transactional operation ( begin transaction and commit )
	 * @param id
	 * @return
	 * @throws IOException 
	 * @throws JsonMappingException 
	 * @throws JsonParseException 
	 */
	Integer updateLock(Integer id) throws JsonParseException, JsonMappingException, IOException;

	/**
	 * Update the given entity <br>
	 * Transactional operation ( begin transaction and commit )
	 * @param id
	 * @param idlock
	 * @return
	 * @throws IOException 
	 * @throws JsonMappingException 
	 * @throws JsonParseException 
	 */
	Integer updateUnlock(Integer id) throws JsonParseException, JsonMappingException, IOException;

	/**
	 * Update the given entity <br>
	 * Transactional operation ( begin transaction and commit )
	 * @param type
	 * @return
	 * @throws IOException 
	 * @throws JsonMappingException 
	 * @throws JsonParseException 
	 */
	Type update(Integer id, Type type) throws JsonParseException, JsonMappingException, IOException;

	/**
	 * Update the given entity <br>
	 * Transactional operation ( begin transaction and commit )
	 * @param type
	 * @return
	 * @throws IOException 
	 * @throws JsonMappingException 
	 * @throws JsonParseException 
	 */
	Type updateWithLock(Integer id, Type type) throws JsonParseException, JsonMappingException, IOException;

	/**
	 * Update the given entity <br>
	 * Transactional operation ( begin transaction and commit )
	 * @param type
	 * @return
	 * @throws IOException 
	 * @throws JsonMappingException 
	 * @throws JsonParseException 
	 */
	Type updateForDelete(Integer id, Integer version) throws JsonParseException, JsonMappingException, IOException;

	/**
	 * Update the given entity <br>
	 * Transactional operation ( begin transaction and commit )
	 * @param type
	 * @return
	 * @throws IOException 
	 * @throws JsonMappingException 
	 * @throws JsonParseException 
	 */
	Type updateForDeleteWithLock(Integer id, Integer version) throws JsonParseException, JsonMappingException, IOException;

	
	/**
	 * Deletes the given entity <br>
	 * Transactional operation ( begin transaction and commit )
	 * @param type
	 * @return true if found and deleted, false if not found
	 */
	void delete(Type type);

	/**
	 * Deletes the entity by its Primary Key <br>
	 * Transactional operation ( begin transaction and commit )
	 * @param id
	 * @return true if found and deleted, false if not found
	 */
	void deleteById(Integer id);

	/**
	 * Loads the entity for the given Primary Key <br>
	 * @param id
	 * @return the entity loaded (or null if not found)
	 */
	Type getById(Integer id);

	/**
	 * Loads ALL the entities (use with caution)
	 * @return
	 */
	List<Type> listAll();

	/**
	 * Count all the occurrences
	 * @return
	 */
	long countAll();

	/**
	 * Check exist
	 * @return
	 */
	boolean isExist(Integer id);

	/**
	 * Loads ALL the entities for select.
	 * @return
	 */
	List<Map<String, Object>> listForSelect();

	/**
	 * Loads ALL the entities for select.
	 * @return
	 */
	List<Map<String, Object>> listForSelectByScope(String scope);

	/**
	 * Search by multiple criteria.
	 * @return
	 */
	List<Type> listWithCriterias(List<SearchCriteria> searchCriterias);

	/**
	 * Search by multiple criteria.
	 * @return
	 */
	List<Type> listWithCriteriasByScope(String scope, List<SearchCriteria> searchCriterias);

	/**
	 * Search by multiple criteria by page.
	 * @return
	 */
	Page<Type> listWithCriteriasByPage(List<SearchCriteria> searchCriterias, Pageable pageable);

	/**
	 * Search by multiple criteria by page.
	 * @return
	 */
	Page<Type> listWithCriteriasByScopeAndPage(String scope, List<SearchCriteria> searchCriterias, Pageable pageable);
	
}
